### ISSUE TITLE
3D images and visualizations

### ISSUE THREAD

**Issue by drgarg** [raghakot/keras-vis#12]
What does one need to do to use these functions on a convnet trained on 3D images using 3D convolutions?

**Comment by raghakot** [comment]
It would require a few all arpund changes in the lib, but shoudl be too difficult. Let me try and take a stab at it today.

**Comment by deepersystems** [comment]
https://github.com/deepersystems/bg/blob/master/nnutil/viz.py

**Comment by gidonro** [comment]
@raghakot did you manage to implement the changes for the 3d visualization? 
Thank you

**Comment by raghakot** [comment]
Yes. As i was making this change, one thing led to another and I ended up making a bunch of efficiency improvements all over the code base. I am planning to get all of this out as one nice release.

**Comment by gidonro** [comment]
Great, thanks!

**Comment by raghakot** [comment]
All visualizations now generalize to N-dim image inputs :D
If you get a chance to use it to visualize your 3D image input, please consider adding your example to examples/ folder. Let me know if you find issue with 3D image visualizations. I didnt get a chance to test them with a pretrained model since i dont have access to one.

**Comment by makquel** [comment]
@raghakot I've came across a PR from someone that made a minor changes to the saliency.py to deal with 3D images, however, I didn't solve it either. As you point it out, now you'd be able to visualize a 3D saliency map, but I got some errors form TensorFlow back end. I'm trying this approach:
```python
grads = visualize_cam(model, 14, filter_indices=20,\
                              seed_input=x_val[0], backprop_modifier='guided')
```
where x_val[0] is a 3D volume (same size as input layer)
